puan/__init__.py

```diff
@@ -45,6 +45,12 @@ class Bounds:
     lower: int = default_min_int
     upper: int = default_max_int
 
+    def __init__(self, lower: int, upper: int):
+        if lower > upper:
+            raise ValueError
```

Comment:
Maybe just a description of what kind of value error?
```suggestion
            raise ValueError(f"upper bound must be higher than lower bound, got: ({lower}, {upper})")
```